File: Haskell/Problem_028.hs
-- What is the sum of the numbers on the diagonals in a 1001 by 1001 spiral?
import           Data.List                      ( foldl' )

fourCorners :: Integer -> Integer
fourCorners n = 4 * n ^ 2 - 6 * (n - 1)

sumDiagonalsSpiral :: Integer -> Integer
sumDiagonalsSpiral side = sum $ 1 : [ fourCorners n | n <- [3, 5 .. side + 1] ]


main :: IO ()
main = do
  print $ sumDiagonalsSpiral 1001

File: Python 3/Problem_028.py
# What is the sum of the numbers on the diagonals in a 1001 by 1001 spiral?


def four_corners(n):
    return (n ** 2) * 4 - (n - 1) * 6


def sum_diagonals_spiral(side):
    diagonals = 1
    for i in range(3, side + 1, 2):
        diagonals += four_corners(i)
    return diagonals


if __name__ == "__main__":
    print(sum_diagonals_spiral(1001))
